src/rust/engine/graph/src/lib.rs

```diff
@@ -382,6 +382,42 @@ impl<N: Node> InnerGraph<N> {
       .filter_map(move |eid| self.entry_for_id(eid))
       .filter_map(move |entry| entry.peek(&context).map(|i| (entry.node(), i)))
   }
+
+  fn prune_unreacheable_from_relevant(&mut self) {
+    let dur = Duration::new(60, 0);
```

Comment:
I expect that expiring things that are younger than 60 seconds is too frequent. Two thoughts on this:
* It should probably be increased to something like 10 minutes.
* This should probably skip entries which are currently running (`self.unsafe_entry_for_id(id).is_running()`).

Also: would use longer, self-explanatory variable names. Rust doesn't really subscribe to the "shortest possible variable names" philosophy of Go, except for trivial cases like `k`/`v` or for parameters in closures.

Reply:
Maybe for now the best thing would be to make this a configurable parameter

Reply:
Alternatively, you could lean in on using the sizes of values to decide what your target size is. It won't be incredibly accurate, but: the [memory stats](https://www.pantsbuild.org/docs/reference-stats#memory_summary) that we already have are generated using the `deepsize` crate: https://github.com/pantsbuild/pants/blob/e53674063f87f9f197573eab80ee30ae2922d132/src/rust/engine/src/scheduler.rs#L145-L173

That might be better for a followup though.

Reply:
Maybe the engine should also emit the memory sizes as an observation histogram? Then the histogram might help with analysis of what the optimal size is.

Reply:
The size is probably a better way to do this long term, but for now duration is easier, so I'll try a follow-up for that.

However, I think I would like to make this a parameter, so I'll need to find a way to configure that and expose it to the engine?  Do we have engine parameters like this?